src/Sylius/Bundle/ProductBundle/Controller/ProductSlugController.php

```diff
@@ -13,17 +13,38 @@
 
 namespace Sylius\Bundle\ProductBundle\Controller;
 
+use Sylius\Component\Product\Generator\SlugGeneratorInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
 class ProductSlugController extends AbstractController
 {
+    private ?SlugGeneratorInterface $slugGenerator;
+
+    public function __construct(?SlugGeneratorInterface $slugGenerator = null)
+    {
+        $this->slugGenerator = $slugGenerator;
+
+        if ($this->slugGenerator === null) {
+            @trigger_error(sprintf('Not passing a $slugGenerator to %s constructor is deprecated since Sylius 1.11 and will be prohibited in Sylius 2.0.', self::class), \E_USER_DEPRECATED);
+        }
+    }
```

Comment:
Why did you add this service in this PR?

Reply:
because
![Screenshot 2021-12-02 at 10 39 50](https://user-images.githubusercontent.com/29897151/144396677-4bdd2b39-3ae9-462f-b32b-f3bb3c89e31c.png)
 in Symfony 5.4 function `get` is deprecated